assets/arc-0060/types.ts

```diff
@@ -0,0 +1,33 @@
+export enum ScopeType {
+    ARBITRARY,
+    LSIG,
+  }
+
+export type StdData = string
```

Comment:
I would avoid strings as they can have a lot of different representations. I would use `Uint8Array`

Reply:
`strings` types are supported by [JSON Schema](https://json-schema.org/understanding-json-schema/reference/type). Having bytes would make ARC-60 interoperability less straightforward.

Note that this approach follows the same rationale as in [ARC-1](https://github.com/algorandfoundation/ARCs/blob/main/ARCs/arc-0001.md#rationale)

Reply:
But strings can be encoded, dynamic sizing and there come be ways of going around the schema verification if you double encode the data to be signed. for example. 

I would highly discourage using string to represent "raw" bytes. 

Instead, we can use JSONSchema to enforce byte level value range and how many exactly we want to sign. For example, a 32 byte challenge for the purpose of auth; could be represented with: 

```json
{
    "$schema": "http://json-schema.org/draft-07/schema#",
    "$id": "http://arc60/schemas/auth.request.json",
    "type": "object",
    "additionalProperties": false,
    "properties": {
        "0": {
            "type": "integer",
            "minimum": 0,
            "maximum": 255
        },
        "1": {
            "type": "integer",
            "minimum": 0,
            "maximum": 255
        },
        "2": {
            "type": "integer",
            "minimum": 0,
            "maximum": 255
        },
        "3": {
            "type": "integer",
            "minimum": 0,
            "maximum": 255
        },
        "4": {
            "type": "integer",
            "minimum": 0,
            "maximum": 255
        },
        "5": {
            "type": "integer",
            "minimum": 0,
            "maximum": 255
        },
        "6": {
            "type": "integer",
            "minimum": 0,
            "maximum": 255
        },
        "7": {
            "type": "integer",
            "minimum": 0,
            "maximum": 255
        },
        "8": {
            "type": "integer",
            "minimum": 0,
            "maximum": 255
        },
        "9": {
            "type": "integer",
            "minimum": 0,
            "maximum": 255
        },
        "10": {
            "type": "integer",
            "minimum": 0,
            "maximum": 255
        },
        "11": {
            "type": "integer",
            "minimum": 0,
            "maximum": 255
        },
        "12": {
            "type": "integer",
            "minimum": 0,
            "maximum": 255
        },
        "13": {
            "type": "integer",
            "minimum": 0,
            "maximum": 255
        },
        "14": {
            "type": "integer",
            "minimum": 0,
            "maximum": 255
        },
        "15": {
            "type": "integer",
            "minimum": 0,
            "maximum": 255
        },
        "16": {
            "type": "integer",
            "minimum": 0,
            "maximum": 255
        },
        "17": {
            "type": "integer",
            "minimum": 0,
            "maximum": 255
        },
        "18": {
            "type": "integer",
            "minimum": 0,
            "maximum": 255
        },
        "19": {
            "type": "integer",
            "minimum": 0,
            "maximum": 255
        },
        "20": {
            "type": "integer",
            "minimum": 0,
            "maximum": 255
        },
        "21": {
            "type": "integer",
            "minimum": 0,
            "maximum": 255
        },
        "22": {
            "type": "integer",
            "minimum": 0,
            "maximum": 255
        },
        "23": {
            "type": "integer",
            "minimum": 0,
            "maximum": 255
        },
        "24": {
            "type": "integer",
            "minimum": 0,
            "maximum": 255
        },
        "25": {
            "type": "integer",
            "minimum": 0,
            "maximum": 255
        },
        "26": {
            "type": "integer",
            "minimum": 0,
            "maximum": 255
        },
        "27": {
            "type": "integer",
            "minimum": 0,
            "maximum": 255
        },
        "28": {
            "type": "integer",
            "minimum": 0,
            "maximum": 255
        },
        "29": {
            "type": "integer",
            "minimum": 0,
            "maximum": 255
        },
        "30": {
            "type": "integer",
            "minimum": 0,
            "maximum": 255
        },
        "31": {
            "type": "integer",
            "minimum": 0,
            "maximum": 255
        }
    },
    "required": [
        "0",
        "1",
        "2",
        "3",
        "4",
        "5",
        "6",
        "7",
        "8",
        "9",
        "10",
        "11",
        "12",
        "13",
        "14",
        "15",
        "16",
        "17",
        "18",
        "19",
        "20",
        "21",
        "22",
        "23",
        "24",
        "25",
        "26",
        "27",
        "28",
        "29",
        "30",
        "31"
    ]
}
```

Reply:
Thank you @ehanoc . 

The schema can be used for passing extra information with the signing bytes. For example, in the authentication process you might want to add the verifier domain or additional data. This is the reason why I added to the ARC-60 schema the `additionalProperties = true`.

Regarding your point, the schema has a required property `bytes` that represent the bytes to be signed. This property can be a byte array as you suggest. I updated the schema to reflect that: https://github.com/deanstef/ARCs/blob/7447a0d82b64677267fdd7e404d5bfc0ff54c5cf/assets/arc-0060/simple-schema.json#L8

Reply:
I am resolving this comment now since we implemented an example for `AUTH`. Feel free to re-open it for further discussion.